lang/src/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -205,6 +220,34 @@ private class ExprValueFactoryImpl(override val ion: IonSystem) : ExprValueFacto
     override fun newTimestamp(value: Timestamp): ExprValue =
         TimestampExprValue(ion, value)
 
+    override fun newTime(value: LocalTime, zoneOffset: ZoneOffset?): ExprValue {
+        return TimeExprValue(ion, value, zoneOffset)
+    }
+
+    override fun newTime(hour: Int, minute: Int, second: Int, nano: Int, precision: Int?, tz_minutes: Int?): ExprValue {
+        // Round nanoseconds to the given precision.
+        val nanoWithPrecision = when  {
+            precision == null || precision >= 9 -> nano
+            else -> (BigDecimal(nano * 10.0.pow(-9)).setScale(precision, RoundingMode.HALF_UP) * 10.0.pow(9)).toInt()
+        }
+        // If the nanos are added to form up a whole second because of the specified precision, carry over the second all up to the hour
+        // and use the mod values of all the new fields to fit in the valid range.
+        val newNano = nanoWithPrecision % 10.0.pow(9).toInt()
+        val newSecond = second + (nanoWithPrecision / 10.0.pow(9).toInt())
+        val newMinute = minute + (newSecond / 60)
+        val newHour = (hour + (newMinute / 60))
+        val localTime = LocalTime.of(
+            newHour % 24,
+            newMinute % 60,
+            newSecond % 60,
+            newNano
+        )
```

Comment:
I am assuming you mean using the java's `LocalTime` APIs. There are bunch of APIs provided which can achieve this but there's one edge case which is not handled by any of the provided APIs. i.e. when the `nano`s round up to form a whole second. For e.g., `TIME (2) '23:59:59.9999'`. In this case,  `nano`s after being rounded overflows and carries over up to the `hour` value. I was considering using `withNanos` API but it will fail for the above mentioned example as the `nano` value overflows.